src/client/api/json_parser/output_foundry.c

```diff
@@ -2,12 +2,169 @@
 // SPDX-License-Identifier: Apache-2.0
 
 #include "client/api/json_parser/output_foundry.h"
+#include "client/api/json_parser/feat_blocks.h"
+#include "client/api/json_parser/native_tokens.h"
+#include "client/api/json_parser/unlock_conditions.h"
+#include "core/models/outputs/output_foundry.h"
 
+/*
+  "outputs": [
+    { "type": 5,
+      "amount": 10000000,
+      "nativeTokens": [],
+      "serialNumber": 123456,
+      "tokenTag": "TokenTAGDemo",
+      "circulatingSupply": "30000000000000000000000000000000000000000",
+      "maximumSupply": "20000000000000000000000000000000000000000",
+      "tokenScheme": 0,
+      "unlockConditions": [
+        {  "type": 0,
+           "address": {
+            "type": 8,
+            "address": "194eb32b9b6c61207192c7073562a0b3adf50a7c"
+            }
+        }
+      ],
+      "featureBlocks": [
+        {
+          "type": 2,
+          "data": "010203040506070809"
+        }
+      ]
+    }
+  ]
+*/
 int json_output_foundry_deserialize(cJSON *output_obj, transaction_essence_t *essence) {
   if (output_obj == NULL || essence == NULL) {
     printf("[%s:%d]: Invalid parameters\n", __func__, __LINE__);
     return -1;
   }
 
-  return 0;
+  int result = -1;
+
+  uint256_t *circ_supply = NULL;
+  uint256_t *max_supply = NULL;
+  native_tokens_t *tokens = native_tokens_new();
+  cond_blk_list_t *cond_blocks = cond_blk_list_new();
+  feat_blk_list_t *feat_blocks = feat_blk_list_new();
+  output_foundry_t *output = NULL;
+
+  // amount
+  uint64_t amount;
+  if (json_get_uint64(output_obj, JSON_KEY_AMOUNT, &amount) != JSON_OK) {
+    printf("[%s:%d]: getting %s json uint64 failed\n", __func__, __LINE__, JSON_KEY_AMOUNT);
+    goto end;
+  }
+
+  // native tokens array
+  if (json_native_tokens_deserialize(output_obj, &tokens) != 0) {
+    printf("[%s:%d]: parsing %s object failed \n", __func__, __LINE__, JSON_KEY_NATIVE_TOKENS);
+    goto end;
+  }
+
+  // serial number
+  uint32_t serial_number;
+  if (json_get_uint32(output_obj, JSON_KEY_SERIAL_NUMBER, &serial_number) != JSON_OK) {
+    printf("[%s:%d]: getting %s json uint32 failed\n", __func__, __LINE__, JSON_KEY_SERIAL_NUMBER);
+    goto end;
+  }
+
+  // token tag
+  byte_t token_tag[TOKEN_TAG_BYTES_LEN];
+  if (json_get_string(output_obj, JSON_KEY_TOKEN_TAG, (char *)token_tag, TOKEN_TAG_BYTES_LEN) != JSON_OK) {
+    printf("[%s:%d]: getting %s json string failed\n", __func__, __LINE__, JSON_KEY_TOKEN_TAG);
+    goto end;
+  }
+
+  // circulation supply
+  char circ_supply_str[STRING_NUMBER_MAX_CHARACTERS];
+  if (json_get_string(output_obj, JSON_KEY_CIRC_SUPPLY, circ_supply_str, STRING_NUMBER_MAX_CHARACTERS) != JSON_OK) {
+    printf("[%s:%d]: getting %s json string failed\n", __func__, __LINE__, JSON_KEY_CIRC_SUPPLY);
+    goto end;
+  }
+  circ_supply = uint256_from_str(circ_supply_str);
+
+  // maximum supply
+  char max_supply_str[STRING_NUMBER_MAX_CHARACTERS];
```

Comment:
for reducing the stack size, both circ and max supply strings should use the same buffer.

Reply:
Fixed.